crates/adapters/src/controller/mod.rs

```diff
@@ -666,6 +679,9 @@ impl Controller {
                 }
                 PipelineState::Terminated => {
                     circuit.kill().map_err(|_| ControllerError::dbsp_panic())?;
+                    for reply_cb in profile_request_receiver.try_iter() {
+                        reply_cb(GraphProfile::default());
```

Comment:
Maybe it's better to return an error in this case?

Reply:
Good idea, I redid the API to use a `Result<GraphProfile, ControllerError>` instead of a default.